modules/social_features/social_embed/src/Form/EmbedConsentForm.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Drupal\social_embed\Form;
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * The form for different setting about embed consent.
+ */
+class EmbedConsentForm extends ConfigFormBase {
+
+  /**
+   * Config settings.
+   *
+   * @var string
+   */
+  const SETTINGS = 'social_embed.settings';
+
+  /**
+   * The cache services.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected CacheBackendInterface $cacheRender;
+
+  /**
+   * Constructs a \Drupal\system\ConfigFormBase object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cacheRender
+   *   The cache services.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, CacheBackendInterface $cacheRender) {
+    parent::__construct($config_factory);
+    $this->cacheRender = $cacheRender;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('cache.render'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'social_embed_admin_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      static::SETTINGS,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config(static::SETTINGS);
+
+    // Add an introduction text to explain what can be done here.
+    // Add warning message.
+    $form['warning_message'] = [
```

Comment:
Yeah lets not do this, we don't want to scare off the SM. Also this is privacy related so if they need to change this it's usually for a critical decision. :D